test/e2e/simple_test.go

```diff
@@ -120,3 +120,46 @@ func TestFailingBuild(t *testing.T) {
 		t.Fatalf("watchBuild did not return expected error: %v", err)
 	}
 }
+
+// TestPodAffinity tests that a build with non existent pod affinity does not scheduled
+// and fails after watch timeout
+func TestPodAffinity(t *testing.T) {
+	clients := setup(t)
+
+	buildName := "affinity-build"
+	if _, err := clients.buildClient.builds.Create(&v1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: buildTestNamespace,
+			Name:      buildName,
+		},
+		Spec: v1alpha1.BuildSpec{
+			Affinity: &corev1.Affinity{
+				NodeAffinity: &corev1.NodeAffinity{
+					//	PreferredDuringSchedulingIgnoredDuringExecution: []
+					RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+						NodeSelectorTerms: []corev1.NodeSelectorTerm{
+							corev1.NodeSelectorTerm{
+								MatchExpressions: []corev1.NodeSelectorRequirement{
+									corev1.NodeSelectorRequirement{
+										Key:      "kubernetes.io/e2e-az-name",
```

Comment:
Can you describe in words why this affinity makes the build unscheduleable?

Reply:
Done